libraries/AP_Compass/Compass.cpp

```diff
@@ -349,13 +349,18 @@ Compass::_detect_backends(void)
         return;
     }
 
-#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX && CONFIG_HAL_BOARD_SUBTYPE != HAL_BOARD_SUBTYPE_LINUX_NONE
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX && CONFIG_HAL_BOARD_SUBTYPE != HAL_BOARD_SUBTYPE_LINUX_NONE && CONFIG_HAL_BOARD_SUBTYPE != HAL_BOARD_SUBTYPE_LINUX_BEBOP
     _add_backend(AP_Compass_HMC5843::detect);
-    _add_backend(AP_Compass_AK8963::detect);
+    _backends[_backend_count++] = new AP_Compass_AK8963(*this,
+                                    new AP_AK8963_SerialBus_MPU9250());
```

Comment:
So... as I said on previous comment: you are unconditionally incrementing _backend_count with a sensor that may not be there.